backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/incremental/utils/Geometry.java

```diff
@@ -0,0 +1,128 @@
+/*******************************************************************************
+ * Copyright (c) 2021 THALES GLOBAL SERVICES.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.diagrams.layout.incremental.utils;
+
+import java.util.Optional;
+
+import org.eclipse.sirius.web.diagrams.Position;
+
+/**
+ * A Geometry Helper.
+ *
+ * @author fbarbin
+ */
+public final class Geometry {
+
+    /**
+     * Computes intersection points between the given rectangle (Bounds) and the segment(source,target), then returns
+     * the closest one from the source point.
+     *
+     * @param source
+     *            the source point of the segment.
+     * @param target
+     *            the target point of the segment.
+     * @param nodeBounds
+     *            the node bounds.
+     * @return the closest intersection from the source point.
+     */
+    public Optional<Position> getIntersection(Position source, Position target, Bounds nodeBounds) {
```

Comment:
Why is that method returning an optional?

Reply:
done